src/main/java/it/gov/pagopa/pu/migration/wf/service/ingestion/debtposition/DebtPositionProcessingService.java

```diff
@@ -0,0 +1,147 @@
+  }
+
+  @Override
+  protected boolean consumeRow(long lineNumber, InstallmentIngestionFlowFileDTO dto, DebtPositionMigrationFileResult migrationFileResult, List<DebtPositionErrorDTO> errorList, Uploads uploads) {
+    String fileName = uploads.getFileName();
+    InstallmentIngestionFlowFileRequiredFieldsValidator.setDefaultValues(dto);
+    if (migrationFileResult.getLastCsvWriter() == null) {
+      DebtPositionErrorDTO error = buildErrorDto(null, lineNumber, "CSV_WRITER_NULL", "CsvWriter not initialized");
+      errorList.add(error);
+      log.error("CsvWriter not initialized at row {}", lineNumber);
+      return false;
+    }
+    StatefulBeanToCsv<InstallmentIngestionFlowFileDTO> csvWriter = migrationFileResult.getLastCsvWriter();
+    try {
+      csvWriter.write(dto);
+      migrationFileResult.setNumCorrectlyProcessedRows(migrationFileResult.getNumCorrectlyProcessedRows() + 1);
+    } catch (Exception e) {
+      DebtPositionErrorDTO error = buildErrorDto(fileName, lineNumber, "CSV_WRITE_ERROR", e.getMessage());
+      errorList.add(error);
+      log.error("Error writing row {} in file {}: {}", lineNumber, fileName, e.getMessage(), e);
+      return false;
+    }
+    return true;
+  }
+
+  @Override
+  protected void setNumTotalRows(DebtPositionMigrationFileResult result, long numTotalRows) {
+    result.setNumTotalRows(numTotalRows);
+  }
+
+  @Override
+  protected void setNumCorrectlyProcessedRows(DebtPositionMigrationFileResult result, long numCorrectlyProcessedRows) {
+    result.setNumCorrectlyProcessedRows(numCorrectlyProcessedRows);
+  }
+
+  @Override
+  protected void setErrorDescription(DebtPositionMigrationFileResult result, String errorDescription) {
+    result.setErrorDescription(errorDescription);
+  }
+
+  @Override
+  protected void setDiscardedFileName(DebtPositionMigrationFileResult result, String discardedFileName) {
+    result.setDiscardedFileName(discardedFileName);
+  }
+
+  /**
+   * Processes a debt position CSV file by streaming its rows, mapping each row to a DTO,
+   * and writing the processed data to a new CSV file using a StatefulBeanToCsv writer.
+   * This avoids loading the entire file into memory.
+   *
+   * @param file      the path to the input CSV file to process
+   * @param uploads   the upload metadata associated with the file
+   * @param errorList the list to which any processing errors will be added
+   * @return a {@link DebtPositionMigrationFileResult} with the results of the processing
+   */
+  public DebtPositionMigrationFileResult processDebtPositionFile(Path file,
+                                                                Uploads uploads,
+                                                                List<DebtPositionErrorDTO> errorList) {
+    Path workingDirectory = file.getParent();
+    String fileName = file.getFileName().toString();
+    Path csvFilePath = workingDirectory.resolve("parsed-" + fileName.replaceFirst("\\.[^.]+$", "") + ".csv");
```

Comment:
the retrieved files are supposed to be csv? you should also configure the final version
```suggestion
    Path csvFilePath = workingDirectory.resolve("parsed-" + fileName.replaceFirst("\\.csv",  "_2_0.csv");
```